src/scripts/services/notifications/dimItemNotificationGenerator.factory.js

```diff
@@ -0,0 +1,83 @@
+const angular = require('angular');
+const _ = require('underscore');
+
+function shuffle(array) {
+  var currentIndex = array.length, temporaryValue, randomIndex;
+
+  // While there remain elements to shuffle...
+  while (currentIndex !== 0) {
+
+    // Pick a remaining element...
+    randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex -= 1;
+
+    // And swap it with the current element.
+    temporaryValue = array[currentIndex];
+    array[currentIndex] = array[randomIndex];
+    array[randomIndex] = temporaryValue;
+  }
+
+  return array;
+}
+
+(function() {
+  'use strict';
+
+  angular.module('dimApp')
+    .factory('dimItemNotificationGenerator', ItemNotificationGenerator)
+    .run(['dimItemNotificationGenerator', (dimItemNotificationGenerator) => {
+      // Assuming on for prototype
+      dimItemNotificationGenerator.start();
+    }]);
+
+  function ItemNotificationGenerator($rootScope, dimNotificationsService) {
+    const createNotification = dimNotificationsService.registerNotificationGenerator({
+      name: "Item Notification Generator",
+      description: "I tell people when they get shiny new loot."
+    });
+    let cancelReloadListener;
+    const alreadyNotified = [];
+
+    // Ideally we'd start/stop this as the user turns it on/off to avoid unneeded processing.
+    return {
+      active: false,
+      start: () => {
+        var self = this;
+        function checkForNewStuff(store) {
+          if (!store) {
+            return;
+          }
+          const newItems = shuffle(_.filter(store.items, function(item) { return !item.notransfer && item.equipment && !item.classified && _.indexOf(alreadyNotified, item.id) < 0; }));
```

Comment:
really cool idea!

curious here, why shuffle? 

btw - we also already have a way to detect if an item is new, could we link into that?
https://github.com/DestinyItemManager/DIM/blob/dev/src/scripts/services/dimStoreService.factory.js#L872